concepts/50 React Components/40 Component Configuration Syntax/55 Markup Customization/3 Using a Rendering Function.md

```diff
@@ -1,42 +1,71 @@
-In the following code, rendering functions are used to specify the [List](https://js.devexpress.com/Demos/WidgetsGallery/Demo/List/ListEditingAndAPI/React/Light)'s [itemTemplate](/api-reference/10%20UI%20Components/dxList/1%20Configuration/itemTemplate.md '/Documentation/ApiReference/UI_Components/dxList/Configuration/#itemTemplate') and the [Button](https://js.devexpress.com/Demos/WidgetsGallery/Demo/Button/PredefinedTypes/React/Light)'s [template](/api-reference/10%20UI%20Components/dxButton/1%20Configuration/template.md '/Documentation/ApiReference/UI_Components/dxButton/Configuration/#template'):
+The following example demonstrates a [DataGrid](/Documentation/Guide/UI_Components/DataGrid/Getting_Started_with_DataGrid/) component which cells display a combination of grid's data and a [TextBox](/Documentation/Guide/UI_Components/TextBox/Getting_Started_with_TextBox/) value. For this task, a rendering function wrapped in [useCallback](https://react.dev/reference/react/useCallback) was used. This way you can pass the TextBox value to the column's [cellRender](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#cellRender) property.
+
+[note] You can also define the rendering function outside the component when you don't need to pass any dependencies to the *'render'* property. 
 
     <!-- tab: Function component -->
-    import List from 'devextreme-react/list';
-    import Button from 'devextreme-react/button';
+    import React, { useState, useCallback } from 'react';
+    import TextBox from 'devextreme-react/text-box';
+    import DataGrid, { Column } from 'devextreme-react/data-grid';
 
-    const renderListItem = (itemData) => {
-        return <p>{itemData.itemProperty}</p>;
-    }
-    const renderButton = (button) => {
-        return <div style={{ padding: 20 }}><p>{button.text}</p></div>;
-    }
+    const dataSource = [1, 2, 3, 4, 5];
+
+    function App() {
+        const [value, setValue] = useState('.');
+
+        const onTextBoxValueChange = (e) => {
+            setValue(e);
+        };
+
+        const cell = useCallback((e) => <div>{e.data + value}</div>, [value]);
```

Comment:
e name is usually used for even arguments. But a template is not an event handler (in its general meaning). I would expect something like this: 

```
const cellRender = useCallback((cell) => <div>{cell.data + value}</div>, [value]);
````